src/app/modules/registration/pages/obs-location/obs-location.page.ts

```diff
@@ -44,35 +46,43 @@ export class ObsLocationPage implements OnInit, OnDestroy {
 
   async ngOnInit() {
     const id = this.activatedRoute.snapshot.params['id'];
+    const geoHazard = this.activatedRoute.snapshot.params['geoHazard'];
+    const lat = this.activatedRoute.snapshot.queryParams['lat'];
+    const lon = this.activatedRoute.snapshot.queryParams['lon'];
+    const locationId = this.activatedRoute.snapshot.queryParams['locationId'];
     if (id) {
       // Edit an existing draft
       this.draft = await this.draftService.load(id);
       this.geoHazard = this.draft.registration.GeoHazardTID;
-    } else if (this.activatedRoute.snapshot.params['geoHazard']) {
+    } else if (geoHazard) {
       // New draft - will be created later
-      this.geoHazard = parseInt(this.activatedRoute.snapshot.params['geoHazard'], 10);
+      if (isNaN(geoHazard)){
+        const geoHazardCapLetter = geoHazard.charAt(0).toUpperCase() + geoHazard.slice(1);
+        this.geoHazard = parseInt(GeoHazard[geoHazardCapLetter], 10);
+      } else {
+        this.geoHazard = parseInt(geoHazard, 10);
+      }
     }
     if (this.geoHazard == null) {
       // No geohazard found, use app mode
       const userSettings = await firstValueFrom(this.userSettingService.userSetting$);
       this.geoHazard = userSettings.currentGeoHazard[0];
     }
+    if(lat && lon && locationId == undefined){
+      this.setLocationMarker(lat, lon);
+    }
+    if(locationId && lat == undefined && lon == undefined){
+      const location =
+        await firstValueFrom(this.locationService.LocationGet({locationId: locationId})) as ObsLocationViewModel;
+      this.setLocationMarker(location.Latitude, location.Longitude);
+      this.selectedLocation = {
+        Name: location.LocationName || location.LocationDescription,
+        Id: locationId
+      };
+    }
     if (this.hasLocation(this.draft)) {
```

Comment:
Og her `else if (this.hasLocation(this.draft)) { ...`  eller er det noe jeg ikke skjønner? Er det meningen at man skal kunne havne inn i flere av disse if-ene?

Reply:
ifene som jeg opprettet er for å starte en ny observasjon på en lokasjon basert på query params
this.hasLocation har ikke noe med urlene å gjøre egentlig. det er for å finne hvilken lokasjon er lagret i draften altså en hel annen ting. vi kunne ha else if her uansett, ja.